From c83e61b780484996bea9f11510412be646e4e6dc Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Thu, 18 Sep 2014 14:14:04 -0400
Subject: [scsi] megaraid_sas_fusion: correctly pass queue info pointer

Message-id: <1411049668-28841-5-git-send-email-thenzl@redhat.com>
Patchwork-id: 96150
O-Subject: [RHEL7.1 e-stor PATCH 04/28] megaraid_sas_fusion: correctly pass queue info pointer
Bugzilla: 1088523
RH-Acked-by: Ewan Milne <emilne@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

The pointer to the queue info structure is potentially
a 64-bit value, so we should be using the correct macros
to set the values in the init frame.

Signed-off-by: Hannes Reinecke <hare@suse.de>
Acked-by: Kashyap Desai <kashyap.desai@lsi.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit fdc5a97cd8bde59b4c94d0b7ef7f11b163e34246)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/megaraid/megaraid_sas_fusion.c b/drivers/scsi/megaraid/megaraid_sas_fusion.c
index e452f68..d10cdf5 100644
--- a/drivers/scsi/megaraid/megaraid_sas_fusion.c
+++ b/drivers/scsi/megaraid/megaraid_sas_fusion.c
@@ -644,7 +644,10 @@ megasas_ioc_init_fusion(struct megasas_instance *instance)
  /* Convert capability to LE32 */
  cpu_to_le32s((u32 *)&init_frame->driver_operations.mfi_capabilities);
 
- init_frame->queue_info_new_phys_addr_lo = cpu_to_le32((u32)ioc_init_handle);
+ init_frame->queue_info_new_phys_addr_hi =
+  cpu_to_le32(upper_32_bits(ioc_init_handle));
+ init_frame->queue_info_new_phys_addr_lo =
+  cpu_to_le32(lower_32_bits(ioc_init_handle));
  init_frame->data_xfer_len = cpu_to_le32(sizeof(struct MPI2_IOC_INIT_REQUEST));
 
  req_desc =
-- 
1.7.1